### PR TITLE
output-files: create output files inside separated directories

### LINKED ISSUE #3885: jobs get-output-files should split downloaded files into subdirectories
> ```
> ll /home/cleber/avocado/job-results/job-2020-06-04T15.17-9bd5b5e
> total 708
> -rw-rw-r--. 1 cleber cleber     41 Jun  4 15:17 id
> drwxrwxr-x. 2 cleber cleber   4096 Jun  4 15:17 jobdata
> -rw-rw-r--. 1 cleber cleber  15818 Jun  4 15:17 job.log
> -rw-rw-r--. 1 cleber cleber 668480 Jun  4 15:17 results.html
> -rw-rw-r--. 1 cleber cleber   1029 Jun  4 15:17 results.json
> -rw-rw-r--. 1 cleber cleber     36 Jun  4 15:17 results.tap
> -rw-rw-r--. 1 cleber cleber    345 Jun  4 15:17 results.xml
> drwxrwxr-x. 5 cleber cleber   4096 Jun  4 15:17 sysinfo
> -rw-rw-r--. 1 cleber cleber     16 Jun  4 15:18 test2_created
> -rw-rw-r--. 1 cleber cleber     16 Jun  4 15:18 test_created
> drwxrwxr-x. 4 cleber cleber   4096 Jun  4 15:17 test-results
> ```
> 
> I had the expectation that, `test_created` would to to `test-results/1-_tmp_t.sh/` and `test2_created` would go to `test-results/2-_tmp_t2.sh/`. Is this expected now? Yes, no, in the future?

### PR DESCRIPTION
**Pull request by beraldoleal**
This is avoiding conflicts with names and fixes #3885.

### CODE REVIEW

File: avocado/plugins/jobs.py [hunk 1]
```diff
@@ -78,8 +78,7 @@ def _save_stream_to_file(self, stream, filename):
         """
         dirname = os.path.dirname(filename)
         if not os.path.isdir(dirname):
-            LOG_UI.error("%s does not exist. Exiting...", dirname)
-            return exit_codes.AVOCADO_GENERIC_CRASH
+            os.makedirs(dirname, exist_ok=True)
```
Comment (willianrampazzo) [inline avocado/plugins/jobs.py:81]: What is the reason to set `exist_ok=True` here if you checked it in the previous `if`?

Comment (beraldoleal) [reply avocado/plugins/jobs.py:81]: You are absolutely right. Missed that, probably without coffe at the time.

Comment (willianrampazzo) [reply avocado/plugins/jobs.py:81]: I know what you mean :D

Comment (beraldoleal) [reply avocado/plugins/jobs.py]: @willianrampazzo since it was a small fix, I just force pushed here.

Comment (willianrampazzo) [reply avocado/plugins/jobs.py]: Cool, thanks!

**Comment by beraldoleal** [comment]
@clebergnu there is also this one.

**Pull request merged**